CMakeLists.txt

```diff
@@ -128,50 +136,104 @@ else()
+
+# Validat the options
 if (NGRAPH_PLAIDML_ENABLE)
     if (NOT DEFINED PlaidML_DIR)
         message(FATAL_ERROR "Provide the PlaidML SDK directory: -DPlaidML_DIR=<directory>")
     endif()
     message(STATUS "nGraph/PlaidML enabled")
 endif()
 
-# Find and build ngraph
-option(NGRAPH_DISTRIBUTED_ENABLE "Add distributed mode to the CPU backend" FALSE)
-if(NGRAPH_DISTRIBUTED_ENABLE)
-    find_package(MPI REQUIRED)
-    add_definitions(-DNGRAPH_DISTRIBUTED)
-    include_directories(SYSTEM ${MPI_C_INCLUDE_PATH} ${MPI_CXX_INCLUDE_PATH})
-    link_directories(${MPI_C_LIBRARIES} ${MPI_CXX_LIBRARIES})
+if(USE_PRE_BUILT_NGRAPH)
+    if(NOT NGRAPH_ARTIFACTS_DIR)
+        message(
+            FATAL_ERROR 
+            "USE_PRE_BUILT_NGRAPH is ON but NGRAPH_ARTIFACTS_DIR is missing"
+        )
+    endif()
+    # Create absolute path for the directory
+    get_filename_component(
+        NGRAPH_ARTIFACTS_DIR
+        "${NGRAPH_ARTIFACTS_DIR}"
+        ABSOLUTE
+    )
+    if (NOT EXISTS ${NGRAPH_ARTIFACTS_DIR})
+        message(FATAL_ERROR 
+            "nGraph artifacts directory doesn't exist: " ${NGRAPH_ARTIFACTS_DIR} )
+    endif()    
 endif()
-ExternalProject_Add(
-    ext_ngraph
-    GIT_REPOSITORY https://github.com/NervanaSystems/ngraph
-    GIT_TAG aac6ee2a2ba1d1a1aa82a9635389c593943ae91d
-    CMAKE_ARGS 
-        -DNGRAPH_DISTRIBUTED_ENABLE=${NGRAPH_DISTRIBUTED_ENABLE}
-        -DNGRAPH_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/ngraph/ngraph_dist 
-        -DNGRAPH_USE_CXX_ABI=${TensorFlow_CXX_ABI}
-        -DNGRAPH_UNIT_TEST_ENABLE=FALSE
-        -DNGRAPH_TOOLS_ENABLE=${NGRAPH_TOOLS_ENABLE}
-        -DNGRAPH_DEX_ONLY=TRUE         
-        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-        -DNGRAPH_GPU_ENABLE=${NGRAPH_GPU_ENABLE}
-        -DNGRAPH_DEBUG_ENABLE=${NGRAPH_DEBUG_ENABLE}
-        -DNGRAPH_PLAIDML_ENABLE=${NGRAPH_PLAIDML_ENABLE}
-        -DPlaidML_DIR=${PlaidML_DIR}
-    TMP_DIR "${CMAKE_CURRENT_BINARY_DIR}/ngraph/tmp"
-    STAMP_DIR "${CMAKE_CURRENT_BINARY_DIR}/ngraph/stamp"
-    DOWNLOAD_DIR "${CMAKE_CURRENT_BINARY_DIR}/ngraph/download"
-    SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/ngraph/src"
-    BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/ngraph/build"
-    BUILD_COMMAND ${CMAKE_MAKE_PROGRAM} -j ${NUM_MAKE_PROCESSES} ngraph
-    UPDATE_COMMAND ""
-    INSTALL_DIR "${CMAKE_INSTALL_PREFIX}"
-)
 
-set( NGRAPH_INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/ngraph/ngraph_dist)
+if(UNIT_TEST_USE_TF_PREBUILT)
+    if(NOT TF_PRE_BUILT_LOCATION)
+        message(
+            FATAL_ERROR 
+            "UNIT_TEST_USE_TF_PREBUILT is ON but TF_PRE_BUILT_LOCATION is missing"
+        )
+    endif()
+    # Create absolute path for the directory
+    get_filename_component(
+        TF_PRE_BUILT_LOCATION
+        "${TF_PRE_BUILT_LOCATION}"
+        ABSOLUTE
+    )
+    if (NOT EXISTS ${TF_PRE_BUILT_LOCATION})
+        message(FATAL_ERROR 
+            "TensorFlow pre-built directory doesn't exist: " ${TF_PRE_BUILT_LOCATION} )
+    endif()
+
+endif()
```

Comment:
Similarly, we might skip `UNIT_TEST_USE_TF_PREBUILT` and only use `TF_PRE_BUILT_LOCATION`?